drivers/platform/xilinx/irq.c

```diff
@@ -230,66 +230,69 @@ int32_t irq_disable(struct irq_ctrl_desc *desc, uint32_t irq_id)
 }
 
 /**
- * @brief Register a callback to handle the irq events.
+ * @brief Unregisters a generic IRQ handling function.
  * @param desc - The IRQ controller descriptor.
  * @param irq_id - Interrupt identifier.
- * @param callback_desc - Callback descriptor
  * @return SUCCESS in case of success, FAILURE otherwise.
  */
-int32_t irq_register_callback(struct irq_ctrl_desc *desc, uint32_t irq_id,
-			      struct callback_desc *callback_desc);
+static int32_t irq_unregister(struct irq_ctrl_desc *desc, uint32_t irq_id)
```

Comment:
Why would you change the order of the functions(irq_unregister and irq_unregister)? It is hard to see the changes if you do this.

Reply:
Because irq_unregister is called in irq_resgister_callback. An to avoid writing the prototype.

Reply:
irq.c includes irq.h, isn't it?

Reply:
I will make a new pull request for this

Reply:
I will make this commit in an other pull request

Reply:
yes, but the last commit removes the function from there.

Reply:
I edit the pull request description regarding this changes